Modules/Core/Common/include/itkAnatomicalOrientation.h

```diff
@@ -0,0 +1,365 @@
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkAnatomicalOrientation_h
+#define itkAnatomicalOrientation_h
+
+#include "ITKCommonExport.h"
+#include "itkImageBase.h"
+#ifndef ITK_FUTURE_LEGACY_REMOVE
+#  include "itkSpatialOrientation.h"
+#endif
+#include <map>
+#include <string>
+
+namespace itk
+{
+
+/** \class AnatomicalOrientation
+ * \brief A class to represent anatomical orientations and convert between conventions.
+ *
+ *  Defines patient specific anatomical names to the XYZ axes of a 3D image.
+ *
+ * \todo update the details of the enum names in the docs
+ *  Instances hold the patient orientation enum and allow conversion to and from enums, string and direction cosine
+ * matrices. Conversions from a direction cosine matrix is approximated with the orientation of the closes axes.
+ *
+ * \ingroup ITKCommon
+ */
+class ITKCommon_EXPORT AnatomicalOrientation
+{
+public:
+  static constexpr unsigned int Dimension = 3;
+  using DirectionType = typename ImageBase<Dimension>::DirectionType;
+  static constexpr unsigned int ImageDimension = Dimension;
+
+#ifndef ITK_FUTURE_LEGACY_REMOVE
+  using LegacyOrientationType = SpatialOrientationEnums::ValidCoordinateOrientations;
+#endif
+
+  enum class CoordinateEnum : uint8_t
+  {
+    UNKNOWN = 0,
+#ifdef ITK_AMBIBUOUS_TO_COORDINATE_ENUMS
+    Left = 2,
+    Right = 3,
+    Anterior = 4,
+    Posterior = 5,
+    Superior = 8,
+    Inferior = 9,
+#endif
+    RightToLeft = 2, ///< 0b0010
+    LeftToRight = 3,
+    PosteriorToAnterior = 4, ///< front - 0b0100
+    AnteriorToPosterior = 5, ///< back
+    InferiorToSuperior = 8,  ///< above - 0b1000
+    SuperiorToInferior = 9,  ///< bottom
+  };
+
+private:
+  enum class CoordinateMajornessTermsEnum : uint8_t
+  {
+    PrimaryMinor = 0,
+    SecondaryMinor = 8,
+    TertiaryMinor = 16
+  };
+
+  template <CoordinateEnum VPrimary, CoordinateEnum VSecondary, CoordinateEnum VTertiary>
+  static constexpr uint32_t m_OrientationValue =
+    (static_cast<uint32_t>(VPrimary) << static_cast<uint8_t>(CoordinateMajornessTermsEnum::PrimaryMinor)) +
+    (static_cast<uint32_t>(VSecondary) << static_cast<uint8_t>(CoordinateMajornessTermsEnum::SecondaryMinor)) +
+    (static_cast<uint32_t>(VTertiary) << static_cast<uint8_t>(CoordinateMajornessTermsEnum::TertiaryMinor));
+
+  CoordinateEnum
+  GetCoordinateTerm(CoordinateMajornessTermsEnum cmt) const;
+
+public:
+#define ITK_ORIENTATION_ENUM(P, S, T) m_OrientationValue<P, S, T>
+
+
+  enum class ToOrientationEnum : uint32_t
```

Comment:
@dzenanz What do you think of this name "ToOrientationEnum", locally I also have a "FromOrientationEnum"? 

Making it `itk::AnatomicalOrientation::ToOrientationEnum`. You earlier suggestions to use a "typedef" (using) make since too, but the enum name seems a little redundent.

The previous was [itk::SpatialOrientation::ValidCoordinateOrientationFlags](https://itk.org/Doxygen52/html/namespaceitk_1_1SpatialOrientation.html#a8240a59ae2e7cae9e3bad5a52ea3496e).  Also the enums were of the longer form "[ITK_COORDINATE_ORIENTATION_RIP](https://itk.org/Doxygen52/html/namespaceitk_1_1SpatialOrientation.html#a8240a59ae2e7cae9e3bad5a52ea3496eaa5d3197482c4335a63a1ad99d6a9edee)" to add even more length.

Reply:
Since they are inside `AnatomicalOrientation` class, we could remove `Orientation` from the name, so have `ToEnum` and `FromEnum`? The previous super-long names were quite unwieldy.